pkg/dependenciesdistributor/dependencies_distributor.go

```diff
@@ -369,6 +320,17 @@ func (d *DependenciesDistributor) syncScheduleResultToAttachedBindings(binding *
 		return err
 	}
 
+	for _, dependency := range dependencies {
+		var gvr schema.GroupVersionResource
+		if gvr, err = restmapper.GetGroupVersionResource(d.RESTMapper, schema.FromAPIVersionAndKind(dependency.APIVersion, dependency.Kind)); err != nil {
+			return err
+		}
+		if !d.InformerManager.IsHandlerExist(gvr, d.EventHandler) {
+			d.InformerManager.ForResource(gvr, d.EventHandler)
+			d.InformerManager.Start()
```

Comment:
1) Kindly move line 330 after line 332 and add boolean variable to check if `d.InformerManager.Start()` should be invoked.
2) Merge codes(line 323 ~ 332) into codes(line 351 ~ 373).

Reply:
Updated.